---
float64 reward
bool done